pyanaconda/modules/subscription/subscription.py

```diff
@@ -242,5 +435,437 @@ def set_system_purpose_with_task(self, sysroot):
         :return: a DBus path of an installation task
         """
         task = SystemPurposeConfigurationTask(sysroot, self.role, self.sla, self.usage, self.addons)
+        # set system purpose as applied once the task finishes running
+        task.stopped_signal.connect(lambda: self._set_is_system_purpose_applied(True))
+        path = self.publish_task(SUBSCRIPTION.namespace, task)
+        return path
+
+    # subscription
+
+    @property
+    def registered(self):
+        """Return True if the system has been registered, False otherwise.
+
+        :return: if the system can be considered as registered
+        :rtype: bool
+        """
+        return self._registered
+
+    @registered.setter
+    def registered(self, system_registered):
+        """Set that the system has been registered.
+
+        :param bool system_registered: system registered state
+        """
+        self._registered = system_registered
+        self.registered_changed.emit()
```

Comment:
Do you want to log setting of the property similarly to the entitlement_attached.setter ?

Reply:
Fixed.